orchagent/pfcactionhandler.cpp

```diff
@@ -34,30 +41,60 @@ PfcWdActionHandler::PfcWdActionHandler(sai_object_id_t port, sai_object_id_t que
             "PFC Watchdog detected PFC storm on queue 0x%lx port 0x%lx",
             m_queue,
             m_port);
-
-    m_stats = getQueueStats(m_countersTable, sai_serialize_object_id(m_queue));
-    m_stats.detectCount++;
-    m_stats.operational = false;
-
-    updateWdCounters(sai_serialize_object_id(m_queue), m_stats);
 }
 
 PfcWdActionHandler::~PfcWdActionHandler(void)
 {
     SWSS_LOG_ENTER();
 
-    auto finalStats = getQueueStats(m_countersTable, sai_serialize_object_id(m_queue));
-
     SWSS_LOG_NOTICE(
-            "Queue 0x%lx port 0x%lx restored from PFC storm. Tx packets: %lu. Dropped packets: %lu",
+            "Queue 0x%lx port 0x%lx restored from PFC storm.",
             m_queue,
-            m_port,
-            finalStats.txPkt - m_stats.txPkt,
-            finalStats.txDropPkt - m_stats.txDropPkt);
+            m_port);
+}
+
+void PfcWdActionHandler::initCounters(void)
+{
+    SWSS_LOG_ENTER();
+
+    if (!getHwCounters(m_hwStats))
+    {
+        return;
+    }
+
+    auto wdQueueStats = getQueueStats(m_countersTable, sai_serialize_object_id(m_queue));
```

Comment:
minor, getQueueStats and updateWdCounters seem to get/set pfcwd counters. Shall we name them follow same routine, which might make it more understandable?

Reply:
These functions are not introduced in this PR. But I agree that renaming should be done. Can I suggest to do it in a separate PR?

Reply:
Sure.